### PR TITLE
Chat activity backend

### PR DESCRIPTION
**Pull request by smikulas**
Hello, this is the chat activity with Firestore backend tested on 2 emulators.

### CODE REVIEW

**Review by lupusmalus — APPROVED** [review]

File: app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt [hunk 1]
```diff
@@ -1,54 +1,151 @@
...
 import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.ChildEventListener
+import com.google.firebase.database.ktx.getValue
 import java.util.*
 
 
 class ChatActivity : AppCompatActivity() {
-    // the text field to which messages are written
+
+    private val _defaultUsername = "Default User"
```
Comment (kprotopapas) [inline app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt:23]: does this need an underscore? Kotlin naming conventions says it should be reserved for private "copies" of public variables. Here there is no public "defaultUsername" so might as well make it without the underscore no?

File: app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt [hunk 2]
```diff
@@ -1,54 +1,151 @@
...
+
+    /**
+     * Listener for the message update. Updates the message bubbles.
+     */
+    private fun realtimeUpdateListener() {
+
+        messageListener = Database.addMessagesListener(_testingChatId) { snapshot ->
+            val messageData = snapshot.getValue<MessageData>()
+
+            messageData!!
```
Comment (kprotopapas) [inline app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt:117]: i get this is checking if it is null, but if it is going to throw an exception anyway is it necessary? Maybe just let it throw later

**Review by kprotopapas — APPROVED** [review]

**Review by sirpudika — APPROVED** [review]
Why is there no test? This is impossible... At least easy stuff like ChatData and MessageData needs to be tested (takes 5 minutes!)

**Comment by whitewind664** [comment]
Forgot about ChatData and MessageData, but for the other stuff we need a mocked DB to test

File: app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt (hunk 2 above)
Comment (whitewind664) [reply app/src/main/java/com/github/gogetters/letsgo/activities/ChatActivity.kt:117]: I think it doesn't compile if I let it be a nullable version

**Pull request merged**